oracle_cardano/chain/cardano_chain_observer.go

```diff
@@ -105,6 +105,29 @@ func (co *CardanoChainObserverImpl) Start() error {
 		})
 	}()
 
+	go func() {
+		for {
+			select {
+			case <-co.ctx.Done():
+				return
+			case err, ok := <-co.syncer.ErrorCh():
+				if !ok {
+					return
+				}
+
+				co.logger.Error("Syncer fatal error", "chainID", co.config.ChainID, "err", err)
+
+				if err := co.syncer.Close(); err != nil {
```

Comment:
Call Dispose here. Dispose should do both things (close syncer, close db). 
Add atomic is disposed check on dispose
`if atomic.CompareAndSwapUint32(&co.isDisposed, 0, 1) {`